### PR TITLE
Feature: Do not contact

### PR DESCRIPTION
**Pull request by ihor-palii**
Allow for "Do Not Contact" Times on platform
https://app.asana.com/0/1200626743215615/1201238931324195/f

### CODE REVIEW

File: templates/triggers/trigger_schedule.haml [hunk 1]
```diff
@@ -36,6 +36,49 @@
...
+        form.off("submit").on("submit", function(e) {
+          e.preventDefault();
+          const formDataByName = buildFormData(form);
+
+          const flowName = getFLowName('#id_schedule_flow');
+          const modalMessage = $('#confirm-schedule-trigger').find('div.p-6');
+          modalMessage.text(getMessageTemplate().replace('{flowName}', flowName));
+          if (formDataByName.start_datetime && formDataByName.start_datetime.length) {
+            let time = parseInt(formDataByName.start_datetime.split(" ")[1].split(":")[0]);
+            if (time < 8 || time > 21) {
```
Comment (teehamaral) [inline templates/triggers/trigger_schedule.haml:63]: I think here would be `if (time < 8 || time > 20)`, as 21h (9 PM) is the limit

File: templates/triggers/trigger_schedule_in_batch.haml [hunk 2]
```diff
@@ -64,6 +73,13 @@
           .replace('{batchInterval}', formDataByName.batch_interval)
 
         modalMessage.text(message);
+        if ("{{ user_org.do_not_contact_property }}" === "True" && formDataByName.start_datetime && formDataByName.start_datetime.length) {
+          let time = parseInt(formDataByName.start_datetime.split(" ")[1].split(":")[0]);
+          if (time < 8 || time > 21) {
```
Comment (teehamaral) [inline templates/triggers/trigger_schedule_in_batch.haml:78]: I think here would be `if (time < 8 || time > 20)`, as 21h (9 PM) is the limit

**Review by teehamaral — CHANGES_REQUESTED** [review]

**Comment by teehamaral** [comment]
@ihor-palii - Could you please implement the unit tests for this feature (only for Python code)?

**Comment by ihor-palii** [comment]
> @ihor-palii - Could you please implement the unit tests for this feature (only for Python code)?

I have added requested changes

**Review by teehamaral — APPROVED** [review]

**Pull request merged**